src/mapping.jl

```diff
@@ -158,12 +177,42 @@ function meltdata(U::AbstractDataFrame, colgroups_::Vector{Col.GroupedColumn})
     end
 
     df = DataFrame(; collect(zip(vnames, V))...)
-    return MeltedData(U, df, indicators, colmap)
+    return MeltedData(U, df, row_indicators, col_indicators, colmap)
+end
+
+
+function meltdata(U::AbstractVector, colgroups_::Vector{Col.GroupedColumn})
+    colgroups = Set(colgroups_)
+
+    if length(colgroups) != 1 || !isnull(first(colgroups).columns)
+        # if every column is of the same length, treat it as a matrix
+        if length(Set([length(u for u in U)])) == 1
+            return meltdata(cat(2, U...), colgroups_)
+        end
+
```

Comment:
a small point, but could save a couple lines here by using `if ... && return...`, or do you not like that notation?